kpops/utils/gen_schema.py

```diff
@@ -0,0 +1,137 @@
+    :return: Whether component is valid for schema generation
+    :rtype: bool
+    """
+    component_name: str = component.__name__
+    # PipelineComponent always has a field 'type'
+    type: str = component.__fields__.get("type").field_info.default  # type: ignore[union-attr]
+    if not component.__fields__.get("schema_type"):
+        log.warning(f"SKIPPED {component_name}, schema_type is not defined")
+        return False
+    schema_type: ModelField = component.__fields__.get("schema_type")  # type: ignore[assignment]
+    schema_type_default = schema_type.field_info.default
+    if (
+        get_origin(schema_type.type_) is not Literal
+        or len(get_args(schema_type.type_)) != 1
+    ):
+        log.warning(
+            f"SKIPPED {component_name}, schema_type must be a Literal with 1 argument of type str."
+        )
+        return False
+    elif get_args(schema_type.type_)[0] != schema_type_default:
+        log.warning(
+            f"SKIPPED {component_name}, schema_type default value must match Literal arg"
+        )
+        return False
+    elif schema_type_default != type:
+        log.warning(f"SKIPPED {component_name}, schema_type != type.")
+        return False
+    elif schema_type_default in used_fields:
+        log.warning(f"SKIPPED {component_name}, schema_type must be unique.")
+        return False
+    else:
+        used_fields.append(schema_type_default)
+        return True
+
+
+def gen_pipeline_schema(
+    components_module: str | None = None,
+) -> None:
+    """
+    Generate a json schema from the models of pipeline components.
+
+    :param components_module: Python module. Only the classes that inherit from PipelineComponent will be considered.
+    :type components_module: str or None
+    :rtype: None
+    """
+
+    components = tuple(_find_classes("kpops.components", PipelineComponent))
+    if components_module:
+        # record components types used so far
+        used_fields: list[str] = [
+            component.__fields__.get("schema_type").default  # type: ignore [union-attr]
+            for component in components
+        ]
+        custom_components = [
+            component
+            for component in _find_classes(components_module, PipelineComponent)
+            if is_valid_component(used_fields, component)
+        ]
+        components += tuple(custom_components)
+
+    # Create a type union that will hold the union of all component types
+    PipelineComponents = Union[components]  # type: ignore[valid-type]
+
+    AnnotatedPipelineComponents = Annotated[
+        PipelineComponents, Field(discriminator="schema_type")
+    ]
+
+    schema = schema_json_of(
+        Sequence[AnnotatedPipelineComponents],
+        title="kpops pipeline schema",
+        by_alias=True,
+        indent=4,
+    ).replace("schema_type", "type")
+    Console(
+        width=1000  # HACK: overwrite console width to avoid truncating output
+    ).print(schema)
```

Comment:
would regular print work as well? I don't think we need rich console here

Reply:
It would work, good point.

Reply:
done